response_operations_ui/contexts/collection_exercise.py

```diff
@@ -0,0 +1,265 @@
+from flask import url_for
+
+REMINDER_EMAILS = {
+    "reminder": "First reminder",
+    "reminder2": "Second reminder",
+    "reminder3": "Third reminder",
+    "reminder4": "Fourth reminder",
+    "reminder5": "Fifth reminder",
+}
+
+NUDGE_EMAILS = {
+    "nudge_email_0": "First nudge email",
+    "nudge_email_1": "Second nudge email",
+    "nudge_email_2": "Third nudge email",
+    "nudge_email_3": "Fourth nudge email",
+    "nudge_email_4": "Fifth nudge email",
+}
+
+CI_TABLE_LINK_TEXT = {
+    "EQ": {"no_instrument": "Add", "restricted": "View", "has_permission": "Select or Add"},
+    "SEFT": {"no_instrument": "Upload", "restricted": "View", "has_permission": "View or Upload"},
+}
+
+
+def build_ce_context(ce_details: dict, has_edit_permission: bool, locked: bool) -> dict:
+    ce = ce_details["collection_exercise"]
+    events = ce_details["events"]
+    short_name = ce_details["survey"]["shortName"]
+    survey_ref = ce_details["survey"]["surveyRef"]
+    collection_instruments = ce_details["collection_instruments"]
+
+    response_chasing = (
+        _build_response_chasing(ce["id"], ce_details["survey"]["id"]) if ce["state"] in ("LIVE", "ENDED") else None
+    )
+    ci_table = _build_ci_table(
+        collection_instruments,
+        locked,
+        ce_details["survey"]["surveyMode"],
+        short_name,
+        ce["exerciseRef"],
+        has_edit_permission,
+    )
+
+    ce_info_events = _ce_events(ce, events)
```

Comment:
For consistency - _ce_info_events rather than _ce_event
```suggestion
    ce_info_events = _ce_events(ce, events)
```